portal-kernel/src/com/liferay/portal/kernel/util/LocaleUtil.java

```diff
@@ -287,16 +287,29 @@ private Locale _fromLanguageId(
 
 		Locale locale = _fromLanguageId(languageId);
 
-		if (validate && !LanguageUtil.isAvailableLocale(locale)) {
-			if (_log.isWarnEnabled()) {
-				_log.warn(languageId + " is not a valid language id");
-			}
+		if (validate) {
+			boolean languageCode = false;
 
-			if (useDefault) {
-				return _locale;
+			if ((languageId.indexOf(CharPool.UNDERLINE) < 0) &&
```

Comment:
@marianoalvarosaiz There was a part of logic to adjust languageId, and your logic here did not include it. 
`if (languageId.equals("zh-Hans-CN")) {
			languageId = "zh_CN";
		}
		else if (languageId.equals("zh-Hant-TW")) {
			languageId = "zh_TW";
		}
		else {
			languageId = StringUtil.replace(
				languageId, CharPool.MINUS, CharPool.UNDERLINE);
		}`

Reply:
@tinatian already covered by com.liferay.portal.kernel.util.LocaleUtilTest.testFromLanguageIdBCP47()

Reply:
@tinatian it is included when I first call `Locale locale = _fromLanguageId(languageId);` After that I only want to check if languageId is a language code (everything that doesn't include a **-** or **_**) which _zh-Hans-CN_ is not so it is cover by the first method call.